src/cpp/include/openvino/genai/tokenizer.hpp

```diff
@@ -124,17 +124,37 @@ class OPENVINO_GENAI_EXPORTS Tokenizer {
     * @param tokenization_params AnyMap with tokenization parameters, e.g. {{"add_special_tokens", false}, {"max_length", 128}}
     * @return pair of [input_ids, attention_mask]
     */
-    TokenizedInputs encode(const std::string prompt, const ov::AnyMap& tokenization_params = {});
+    TokenizedInputs encode(const std::string& prompt, const ov::AnyMap& tokenization_params = {});
 
     /**
-    * @brief encode batch of prompts. Left padding will be applied by default
+    * @brief encode batch of prompts.
     * @param prompts vector storing batch of prompts
     * @param tokenization_params AnyMap with tokenization parameters, e.g. {{"add_special_tokens", false}, {"max_length", 128}}
     * @return pair of [input_ids, attention_mask]
     */
-    TokenizedInputs encode(std::vector<std::string>& prompt, const ov::AnyMap& tokenization_params = {});
-    TokenizedInputs encode(std::vector<std::string>&& prompts, const ov::AnyMap& tokenization_params = {});
-    TokenizedInputs encode(std::initializer_list<std::string>& prompts, const ov::AnyMap& tokenization_params = {});
+    TokenizedInputs encode(const std::vector<std::string>& prompt, const ov::AnyMap& tokenization_params = {});
+    TokenizedInputs encode(const std::vector<std::string>&& prompts, const ov::AnyMap& tokenization_params = {});
+    TokenizedInputs encode(const std::initializer_list<std::string>& prompts, const ov::AnyMap& tokenization_params = {});
+   
+    /**
+    * @brief encode paired prompts.
+    * 
+    * @param prompts vector storing batch of prompts
+    * @param tokenization_params AnyMap with tokenization parameters, e.g. {{"add_special_tokens", false}, {"max_length", 128}}
+    * @return pair of [input_ids, attention_mask]
+    */
+   TokenizedInputs encode(const std::vector<std::pair<std::string, std::string>>& prompt, const ov::AnyMap& tokenization_params = {});
```

Comment:
Add a comment that this overload is less efficient because it copies the strings to match the signature with two vectors.

An overload with non const && could be added in addition to that one to fix this, but I don't see how to avoid code duplication. This function could be templated to implement the perfect forwarding, but this would replace specific `prompt` type with a template type which is harder to read

Reply:
Added comment about efficiency.